### PR TITLE
Fix: Double-clicking watch expression values causes editing

### LINKED ISSUE #7867: Double-clicking watch expression values causes editing
Reps in other parts, like the Scopes panel, allow double-click to select text. Watch expressions listen on the whole element for double-clicks and switch to edit; even though only the expression should be editable.

This matches behavior for Chrome as well.

@darkwing, possibly a good first bug?

### PR DESCRIPTION
**Pull request by lawrencek0**
Fixes #7867 

### Summary of Changes

* Moved the function `editExpression` to ObjectInspector to pass it to ObjectInspectorItem

### CODE REVIEW

**Comment by princiya** [comment]
@lawrencek0 Please update the Jest snapshot - `yarn test -u` 🙏🏻

**Comment by darkwing** [comment]
This is a great start @lawrencek0 !  Welcome!

One thing I did notice was that if I add the expression `document.body.firstChild`, expand the node, then double-click a property (like `async`), the edit interface appears.  I think we need to isolate the doubleclick-to-edit to *just* the property entered by the user.

File: packages/devtools-reps/src/object-inspector/components/ObjectInspectorItem.js [hunk 1]
```diff
@@ -286,7 +293,8 @@ class ObjectInspectorItem extends Component<Props> {
                 setExpanded: this.props.setExpanded
               });
             }
-          : undefined
+          : undefined,
+        onDoubleClick: onExpressionDblClick
```
Comment (darkwing) [inline packages/devtools-reps/src/object-inspector/components/ObjectInspectorItem.js]: I see that the `renderLabel` function this appears in has a few interesting things we could try:

1.  The `onClick` has a check for selection:  `if (Utils.selection.documentHasSelection()) {` ; maybe that could be of use?

2.  There is a `depth` check; maybe if `depth` is one, and there's more than one depth, we ignore on depth 0?

Comment (lawrencek0) [reply packages/devtools-reps/src/object-inspector/components/ObjectInspectorItem.js]: Thanks @darkwing. I implemented a depth check for that

**Comment by flodolo** [comment]
I have the feeling there were some issues with a rebase? I see changes that I'm not sure belong to this PR, at least in the .properties file

**Comment by lawrencek0** [comment]
I am not sure. I rebased from the origin/master branch. Should I close this PR and open a new one in a new branch?

**Comment by darkwing** [comment]
@lawrencek0 It may be easiest to do such, since your changes were small. 👍

**Comment by lawrencek0** [comment]
Closing for #8013

**Pull request closed without merging**